core/domain/collection_domain.py

```diff
@@ -1030,6 +1030,30 @@ def get_node(self, exploration_id):
                 return node
         return None
 
+    def sort_nodes(self):
+        """Arrange the nodes in topologically sorted order unless it's
+        impossible, otherwise returns the unordered nodes.
```

Comment:
This talks about returning things, but nothing is returned by the method.

There should be a "Raises:" docstring.